js/index.js

```diff
@@ -0,0 +1,62 @@
+let listBooks = [];
+// Local Storage
+const saveToLocalStorage = () => {
+  localStorage.setItem('Library', JSON.stringify(listBooks));
+};
+
+const getDataFromLocalStorage = () => {
+  const data = JSON.parse(localStorage.getItem('Library'));
+  if (data !== null) {
+    listBooks = data;
+  }
+};
+
+function displayBooks() {
+  const section = document.querySelector('#book-list');
+  let books = '';
+  getDataFromLocalStorage();
+
+  listBooks.forEach((book, index) => {
+    books += `<article class="book">
+    <p>${book.title} <br> ${book.author}</p>
+    <button type="button" id="${index}" class="remove-btn" onclick="removeBook(${index})">Remove</button>
+    <hr>
+  </article>`;
+  });
+
+  if (listBooks.length === 0) {
+    books = '<p>Library is empty...</p>';
+  }
+  section.innerHTML = books;
+}
+
+function removeBook(id) {
+  const bookId = parseInt(id, 2);
+  listBooks = listBooks.filter((book, index) => index !== bookId);
+  saveToLocalStorage();
+  displayBooks();
+}
```

Comment:
You've done well so far in your implementation to delete a book, but there are a few changes to be made for the delete process to work fine as expected:

- [ ] Currently, when I add books and attempt to remove the last one or second to last, the book clicked cannot be removed. And also, the removal only happens for books at the top of the list. Kindly redo the deletion of a book so the right book is deleted whenever the remove button is clicked for a book.

- To achieve this, I recommend, when adding a book, the object you created that is pushed to the bookList array, you add the id property for a new book and give it a random id, so when you delete a book with the filter method, you compare the book clicked by the ID of the book filtered.